### DISCUSSION TITLE
Funds automatic management

### DISCUSSION THREAD

**Discussion by iafi01** [MetaMask/snaps#1052] [Ideas]
I think can be great for the wallet user experience, manage automatic pre-set operation just with in the graphic interface, that will be executed on Dexes and Money Markets in Defi without the need of use different defi platform interfaces and without the risk of manual management.

it will be perfect for my team if it's possible to collect a small fee on swaps to cover costs of developing, for this reason we are studying to understand that, in other way we have to sell the metamask plugin.

If someone has suggest for us, it will be really appreciated.

**Comment by ritave** [comment]
Hello @iafi01, I don't quite understand what are you asking for.

Maybe your team would like to propose a new API through [Snaps Improvement Proposals](https://github.com/MetaMask/SIPs) program? One such API that was proposed there and then implemented would be transactions insights in [SIP-3](https://github.com/MetaMask/SIPs/blob/main/SIPS/sip-3.md).

**Comment by iafi01** [comment]
I'll try to explain as simple as possible, today a lot of people use bots to do automatic operations with their crypto, people pay freelance to develop for them a the bot and the interface that seems a wallet, all that can be done with snaps, users give the access to sign transaction to the snap and it manage their funds automatically.

So the question is if already there is the permission to sign transaction without click confirm button in metamask every time, like give an allowance of spend my funds to the snaps

Thank you, regards 
Lorenzo

**Reply by ritave** [reply]
You can easily create a Snap Bot that trades your assets using [`snap_getBip32Entropy`](https://docs.metamask.io/guide/snaps-rpc-api.html#snap-getbip32entropy)
